templates/webhook.bin.erb

```diff
@@ -108,7 +114,13 @@ class Server < Sinatra::Base
     data = JSON.parse(decoded, :quirks_mode => true)
 
     # github sends a 'ref', stash sends an array in 'refChanges'
-    branch = ( data['ref'] || data['refChanges'][0]['refId'] ).sub('refs/heads/', '')
+    if $config['git_provider'] == 'github'
+      branch = ( data['ref'] || data['refChanges'][0]['refId'] ).sub('refs/heads/', '')
```

Comment:
That would work. I was concerned with the logic clause becoming unmanageable over time. Either way actually works for me. I can resubmit.